SU-SWS/decanter-react-forms
# ADAPT-3096 Create Text Field component

# READY FOR REVIEW

# Summary
- Create Text Field component

# Needed By (Date)
- tbd

# Urgency
- medium-high

# Steps to Test

1. Check out branch locally
2. View local preview or [Netlify](https://deploy-preview-1--decanter-react-forms.netlify.app/?path=/story/composite-text-field--default) and confirm the following design tidbits (see [Figma mockup](https://www.figma.com/file/Kmd4utmJFPRMVeCFEEBQhLtx/Decanter-Design-System?node-id=10694%3A292))
- Text should be ~18px
- Input padding should be 19px left, top, right and 20px bottom.
- Icon should have two position options — Left or Right
- Error label, input, and message should be `digital-red-light`
- Valid label, input, and message should be `digital-green`
- Custom Icons should work (see [Custom Input with Icon Preview](https://deploy-preview-1--decanter-react-forms.netlify.app/?path=/story/composite-text-field--user-input))
And check to see if anything else appears out of place. 😄 

**Additional Notes:** 
- Disable state is currently just background color `black-10` since the diagonal stripes cannot be implemented through tailwind utility class styles.
- Currently the Char count is live update. I have not tested it for c&p reactions nor determined how we would handle errors in the event that characters exceed the limit/limit has been reached.
- [Password Error and Validation checklist](https://www.figma.com/file/Jk1dfcFHSA59ljplAUEo2C/My-Account-Wireframes-2021?node-id=1037%3A11547) has not been implemented. I plan to initiate further discussion on how we would like to handle that (either through an existing library (e.g. [react-password-checklist](https://www.npmjs.com/package/react-password-checklist)) or creating a password specific component) during ADAPT Dev Chat 

4. Review the code!
I anticipate that some conditional statements might need to be refactored. Please do not hesitate to tag me, ask any questions, and suggest additional examples/stories to be added. Thank you!!! 🥳 

# Associated Issues and/or People
- [ADAPT-3096](https://stanfordits.atlassian.net/browse/ADAPT-3096)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
